src/transformations/peer-import.ts

```diff
@@ -0,0 +1,36 @@
+import { SourceUnit } from 'solidity-ast';
+import { getNodeBounds } from '../solc/ast-utils';
+import { Transformation } from './type';
+import { TransformerTools } from '../transform';
+
+export function* peerImport(
+  ast: SourceUnit,
+  { getData }: TransformerTools,
+): Generator<Transformation> {
+  for (const node of ast.nodes) {
+    switch (node.nodeType) {
+      case 'ContractDefinition':
+      case 'EnumDefinition':
+      case 'ErrorDefinition':
+      case 'FunctionDefinition':
+      case 'StructDefinition':
+      case 'UserDefinedValueTypeDefinition':
+      case 'VariableDeclaration': {
+        const { importFromPeer } = getData(node);
+        if (importFromPeer !== undefined) {
+          yield {
+            ...getNodeBounds(node),
+            kind: 'replace-declaration-with-peer-import',
+            text: `import { ${node.name} } from "${importFromPeer}";`,
+          };
+        }
+        break;
+      }
+      case 'ImportDirective':
+      case 'PragmaDirective':
+      case 'UsingForDirective': {
+        break;
+      }
+    }
```

Comment:
```suggestion
    const { importFromPeer } = getData(node);
    if (importFromPeer !== undefined) {
      yield {
        ...getNodeBounds(node),
        kind: 'replace-declaration-with-peer-import',
        text: `import { ${node.name} } from "${importFromPeer}";`,
      };
    }
```

Reply:
ok, you rely on `importFromPeer` being set, and we keep the switch in `getExcludeAndImportPathsForPeer`